packages/next-app/src/components/RootLayout.tsx

```diff
@@ -2,14 +2,17 @@
 
 import { darkTheme, RainbowKitProvider, Theme } from "@rainbow-me/rainbowkit";
 import merge from "lodash.merge";
+import { usePathname } from "next/navigation";
 import React from "react";
 
 import { WagmiConfig } from "#/wagmi";
-import { chains, client } from "#/wagmi/client";
+import { getWagmiClient } from "#/wagmi/client";
 
 import { ToastProvider } from "./Toast";
 
 export function RootLayout({ children }: React.PropsWithChildren) {
+  const path = usePathname();
+  const { client, chains } = getWagmiClient(path);
```

Comment:
👍 got it. A comment explaining the rationale behind using this would've been helpful here